apps/scan/backend/src/printing/print_report_section.ts

```diff
@@ -0,0 +1,100 @@
+import memoizeOne from 'memoize-one';
+import { PollsTransitionType } from '@votingworks/types';
+import { assert } from '@votingworks/basics';
+import { isPollsSuspensionTransition } from '@votingworks/utils';
+import {
+  PrecinctScannerBallotCountReport,
+  PrecinctScannerTallyReports,
+} from '@votingworks/ui';
+import { PaperDimensions, renderToPdf } from '@votingworks/printing';
+import { PrintResult } from '@votingworks/fujitsu-thermal-printer';
+import { Store } from '../store';
+import { getMachineConfig } from '../machine_config';
+import { getScannerResults } from '../util/results';
+import { getCurrentTime } from '../util/get_current_time';
+import { rootDebug } from '../util/debug';
+import { Printer } from './printer';
+
+const debug = rootDebug.extend('print-report-section');
+
+async function getReportSections(
+  store: Store,
+  pollsTransitionType: PollsTransitionType,
+  pollsTransitionBallotCount: number,
+  pollsTransitionTime: number
+): Promise<JSX.Element[]> {
+  debug('generating all report sections...');
+  const electionDefinition = store.getElectionDefinition();
+  const precinctSelection = store.getPrecinctSelection();
+  const isLiveMode = !store.getTestMode();
+  const { machineId } = getMachineConfig();
+  assert(electionDefinition);
+  assert(precinctSelection);
+
+  const scannerResultsByParty = await getScannerResults({ store });
+
+  if (isPollsSuspensionTransition(pollsTransitionType)) {
+    return [
+      PrecinctScannerBallotCountReport({
+        electionDefinition,
+        precinctSelection,
+        totalBallotsScanned: pollsTransitionBallotCount,
+        pollsTransition: pollsTransitionType,
+        pollsTransitionedTime: pollsTransitionTime,
+        reportPrintedTime: getCurrentTime(),
+        isLiveMode,
+        precinctScannerMachineId: machineId,
+      }),
+    ];
+  }
+
+  return PrecinctScannerTallyReports({
+    electionDefinition,
+    precinctSelection,
+    isLiveMode,
+    pollsTransition: pollsTransitionType,
+    pollsTransitionedTime: pollsTransitionTime,
+    reportPrintedTime: getCurrentTime(),
+    precinctScannerMachineId: machineId,
+    electionResultsByParty: scannerResultsByParty,
+  });
+}
+
+const getReportSectionsMemoized = memoizeOne(getReportSections);
```

Comment:
I didn't run into perf problems while making this PR, but we did run into perf problems when testing the all bubble ballot. With so many contests, it took almost a minute to create a report for 1,000's of ballots. The expensive part is not the DB fetches or the React initializations, I believe, it's the tabulation itself. The all bubble ballot is an edge case, but I think an informative one. If we had an equivalently large election that was a primary, I'd rather the wait be all up front rather than repeated between each print.